### PR TITLE
refactor startup, better logging, and more predictable startup

### PR DESCRIPTION
**Pull request by herb**

### CODE REVIEW

File: grouper/setup.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    parser.add_argument(
+            "-n", "--deployment-name", type=str, default="", help="Name of the deployment.")
+
+    return parser.parse_args()
+
+
+def setup_logging(args, log_format):
+    # type: (Namespace, str) -> None
+
+    verbose = args.verbose * 10
```
Comment (Acidity) [inline grouper/setup.py]: Might be a good idea to add a comment for why we're multiplying by 10 here

File: bin/grouper-fe [hunk 2]
```diff
@@ -12,96 +12,59 @@ import tornado.ioloop
...
+def main(args, sentry_client):
+    # type: (argparse.Namespace, SentryProxy) -> None
+
+    log_level = logging.getLevelName(logging.getLogger().level)
+    logging.info("begin. log_level={}".format(log_level))
+
+    assert not (settings.debug and settings.num_processes > 1), \
+            "debug mode does not support multiple processes"
+
+    if settings.plugin_dir:
```
Comment (Acidity) [inline bin/grouper-fe]: Should this be a call to load_plugins instead?

File: bin/grouper-api [hunk 3]
```diff
@@ -15,67 +15,48 @@ from grouper.api.settings import settings
...
-        sa_log.setLevel(logging.INFO)
+    log_level = logging.getLevelName(logging.getLogger().level)
+    logging.info("begin. log_level={}".format(log_level))
 
     if settings.plugin_dir:
-        if not os.path.exists(settings.plugin_dir):
-            logging.fatal("Plugin directory does not exist")
-            sys.exit(1)
-        load_plugins(settings.plugin_dir, service_name="grouper-api")
+        logging.debug("loading plugins from {}".format(settings.plugin_dir))
```
Comment (Acidity) [inline bin/grouper-api]: Should this be a call to load_plugins instead?

**Comment by Acidity** [comment]
Other than noted, LGTM

File: bin/grouper-api (hunk 3 above)
Comment (herb) [inline bin/grouper-api]: yes, good catch.

File: bin/grouper-fe (hunk 2 above)
Comment (herb) [inline bin/grouper-fe]: yes, good catch.

File: grouper/setup.py (hunk 1 above)
Comment (herb) [inline grouper/setup.py]: added

File: grouper/setup.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    logging.basicConfig(level=log_level, format=log_format)
+    if log_level < 0:
+        sa_log.setLevel(logging.INFO)
+
+
+def load_plugins(settings, service_name):
+    if settings.plugin_dir:
+        assert os.path.exists(settings.plugin_dir), "Plugin directory does not exist"
+        load_plugins(settings.plugin_dir, service_name=service_name)
```
Comment (Acidity) [inline grouper/setup.py]: Is this intended to be recursive?

Comment (herb) [inline grouper/setup.py]: **_mumble mumble_**

**Pull request merged**